src/content/concepts/module-federation.md

```diff
@@ -109,6 +111,207 @@ This plugin adds specific references to containers as externals and allows to im
 
 This plugin combines `ContainerPlugin` and `ContainerReferencePlugin`. Overrides and overridables are combined into a single list of specified shared modules.
 
+## Shared Libraries and Versioning
+
+### Specify package versions
+
+There are three ways to specify the versions of shared libraries.
+
+**Array syntax**
+
+This syntax allows you to easily share libraries with package name only. This approach is good for prototyping, but it will not allow you to scale to large production environment given libraries like `react` and `react-dom` will require additional requirements.
+
+```js
+module.exports = {
+  plugins: [
+    new ModuleFederationPlugin({
+      // adds lodash as shared module
+      // version is inferred from package.json
+      // there is no version check for the required version
+      // so it will always use the higher version found
+      shared: ['lodash'],
+    }),
+  ],
+};
+```
+
+**Object syntax**
+
+This syntax provides you more control over each shared library in which you can define package name as the key and version ([semver](https://semver.org/)) as the value.
+
+```js
+module.exports = {
+  plugins: [
+    new ModuleFederationPlugin({
+      shared: {
+        // adds lodash as shared module
+        // version is inferred from package.json
+        // it will use the highest moment version that is >= 4.17 and < 5
+        lodash: '^4.17.0',
+      },
+    }),
+  ],
+};
+```
+
+**Object syntax with sharing hints**
+
+This syntax allows you to provide additional [hints](#sharing-hints) to each shared package where you define the package name as the key, and the value as an object containing hints to modify sharing behavior.
+
+```js
+const deps = require('./package.json').dependencies;
+
+module.exports = {
+  plugins: [
+    new ModuleFederationPlugin({
+      shared: {
+        // adds react as shared module
+        // version is inferred from package.json
+        // it will always use the shared version, but print a warning when the shared lodash is < 4.17 or >= 5
+        react: {
+          requiredVersion: deps.lodash,
+          singleton: true,
+        },
+      },
+    }),
+  ],
+};
+```
+
+### Sharing hints
+
+**`eager`**
```

Comment:
Use heading here.

Reply:
Thank you for the review. I updated the heading but I am not sure if it is correct. Please let me know if I need to make additional changes.

Reply:
Thank you taking the time to make the changes!